### ISSUE TITLE
Is this unsupported?

### ISSUE THREAD

**Issue by zilions** [brianc/node-domain-middleware#14]
Seems like it's been a while since the last update. Is it now unsupported?

**Comment by brianc** [comment]
I still use it - it's just pretty simple & doesn't really need much more so
it doesn't really get updated unless I hit a bug with it.

**Comment by zilions** [comment]
Awesome.
So it's still compatible with the latest version of NodeJS?

**Comment by brianc** [comment]
AFAIK, yes

**Comment by avnersorek** [comment]
It might still work but please note that Domains have been officially deprecated - 

> Most end users should not have cause to use this module. Users who absolutely must have the functionality that domains provide may rely on it for the time being but should expect to have to migrate to a different solution in the future

https://nodejs.org/api/domain.html

**Comment by brianc** [comment]
Yeah.  Do you know of a viable alternative to domains? I was under the
impression that was still the best one could get at managing async errors &
keeping context in a long block of CSP code.

**Comment by avnersorek** [comment]
nope :cry: 
I'll probably keep using domains until they remove them altogether and hopefully by then there will be some sort of alternative.

**Comment by ChrisCinelli** [comment]
Domain were softly deprecated more than one year ago. The core node.js developers hate them because of the amount of code necessary around APIs to keep domains working. Some use cases are broken in the latest version of node. But nobody came out with a better alternative.

**Comment by rafis** [comment]
Is it possible to improve this module (perhaps version 1.0) in the way documentation says:

> The better approach is to send an error response to the request that triggered the error, while letting the others finish in their normal time, and stop listening for new requests in that worker.

It says a domains usually causes a resources leak, the proper way would be to stop accepting new connections, wait until current connections are finished, possibly with configurable timeout, and then `process.exit(1);` to allow a supervisor mechanishm to restart an app.
